slides/src/content/rxjs-observables.md

```diff
@@ -1,164 +1,191 @@
+  4. is not required in Angular
 
 ---
 
-## Constructing Observables from Other Sources
+## The Map Operator
 
-You can construct an Observable by converting existing JavaScript constructs like event listeners.  
+* You pass it a function that transforms (maps) the data returned.
 
-For instance, say we have a button defined in our DOM as such:
+For example: an observable based on an input event will pass the event object to the callback.  We can make it return the input's value using the `map` operator:
 
-```html
-<button>Click Me!</button>
+```javascript
+const searchString$ = Observable
+  .fromEvent(document.querySelector('.search-input'), 'input')
+  .map(event => {
+    return event.target.value;
+  });
 ```
 
-And we want `Clicked!` to be printed to console whenever the button is clicked. 
-We can use `fromEvent` creation operator to create an Observable from the click event listener:
+Notes:
+https://jsbin.com/genayadivo/edit?html,js,console,output
+*concepts*: operators, map operator (instance method)
 
-```js 
-import {Observable} from 'rxjs/Observable';
-import 'rxjs/add/observable/fromEvent';
++++
 
-const clickStream = Observable.fromEvent(document.querySelector('button'), 'click');
+## Quiz 3
 
-const clickStreamSubscription = clickStream.subscribe(
-  () => console.log('Clicked!')
-);
+3. The map operator
+  1. will perform `Array.map` on the result of an observable, only when it is an array.
+  2. will return an observable
+  3. enables geolocation in Angular
+  4. will modify the result returned by an observable
+
+---
+
+## Angular's Http Service
+
+* `Http` from `'@angular/http'` has methods such as `get` that return observables.
+* Map is often used with `Http` as in the following example:
+
+```javascript
+const getData$ = http.get('https://example.com/api/products')
+.map(response => {
+  return response.json();
+});
+getData$.subscribe((data) => {
+  // 
+});
 ```
 
-Similarly to `fromEvent`, RxJS library provides [a number of different ways to create Observables](http://reactivex.io/rxjs/manual/overview.html#creation-operators) such as `fromPromise`.
+In this example, `map` is used because `get` is an observable of an object with a `json` method.  Usually, we want the return value of `json()`, not the response object!
 
 ---
 
-## Disposing (Cancelling) Observable Execution
-
-Say, for some reason, we want to the button to stop printing `Clicked!` to console after 3 seconds. 
+## What about Unsubscribing?
 
-We can achieve this by *disposing* or *cancelling* the Observable execution. This is done by *unsubscribing* from the executing Observable.
+* Obserables that finish, including `http.get` don't need to be unsubscribed.
+* If our observable keeps pushing values, the function we gave it when we subscribed will be called until we unsubscribe. We can stop this by unsubscribing from the observable.
+* The `subscribe()` method returns a *subscription*.
+* A subscription has an `unsubscribe()` method.
```

Comment:
A good example to unsubscribe would be to consider netflix like streaming services, where a user decided to browse away from a movie to load another movie.